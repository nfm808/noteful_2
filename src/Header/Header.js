import React from 'react'
import { Link } from 'react-router-dom'
import './Header.css'

export default function Header(props) {
  return (
    <header className="App__header">
      <h1><Link to="/">{props.title}</Link></h1>
    </header>
  )
}
