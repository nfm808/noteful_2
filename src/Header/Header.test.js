import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Header from './Header'

describe('Header component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header title='Noteful' />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Header title='test title' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
