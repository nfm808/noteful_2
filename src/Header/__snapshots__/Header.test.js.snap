// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the UI as expected 1`] = `
<header
  className="App__header"
>
  <h1>
    test title
  </h1>
</header>
`;
